src/App.tsx

```diff
@@ -26,70 +26,111 @@ type ReorderOptions = {
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+  render() {
+    const goodsToRender = getReorderedGoods(goodsFromServer, this.state);
 
-        <button
-          type="button"
-          className="button is-danger is-light"
+    return (
+      <div className="section content">
+        <div
+          className="buttons"
+          role="directory"
         >
-          Reset
-        </button>
-      </div>
+          <button
+            type="button"
+            className={this.state.sortType === SortType.ALPHABET
+              ? 'button is-info' : 'button is-info is-light'}
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET, isReversed: false });
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={this.state.sortType === SortType.LENGTH
+              ? 'button is-success' : 'button is-success is-light'}
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH, isReversed: false });
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed
+              ? 'button is-warning' : 'button is-warning is-light'}
+            onClick={() => {
+              this.setState((state) => ({ isReversed: !state.isReversed }));
+            }}
+          >
+            Reverse
+          </button>
+
+          {(this.state.sortType !== SortType.NONE || this.state.isReversed)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                this.setState({ isReversed: false, sortType: SortType.NONE });
+              }}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {goodsToRender.map(good => {
+              return (
+                <li
+                  data-cy="Good"
+                  key={Math.random() + Math.random()}
```

Comment:
Please, pay attention to the mentors' comments.

<img width="884" alt="image" src="https://user-images.githubusercontent.com/102293862/234918050-9d80f53c-8001-425f-8792-607d69d12362.png">
https://react.dev/learn/rendering-lists#rules-of-keys